test/util/memoizer-test.ts

```diff
@@ -0,0 +1,444 @@
+import Immutable from 'immutable';
+import should from 'should';
+import { Code, Concept, DateTime, Quantity, Ratio, ValueSet } from '../../src/cql';
+import { Uncertainty } from '../../src/datatypes/uncertainty';
+import { equals } from '../../src/util/comparison';
+import * as Memoizer from '../../src/util/memoizer';
+
+describe('Memoizer Tests', () => {
+  let immutableMemoizer: Memoizer.ImmutableMemoizer<unknown, unknown>;
+
+  beforeEach(() => {
+    immutableMemoizer = new Memoizer.ImmutableMemoizer();
+  });
+
+  it('should normalize null and undefined properties', () => {
+    const c1: { code: string; system: undefined } = { code: 'a', system: undefined };
+    const c2: { code: string; system: null } = { code: 'a', system: null };
+
+    const ic1 = immutableMemoizer.toImmutableObjectKey(c1);
+    const ic2 = immutableMemoizer.toImmutableObjectKey(c2);
+
+    equals(c1, c2).should.be.true();
+    equals(c2, c1).should.be.true();
+    Immutable.is(ic1, ic2).should.be.true();
+    Immutable.is(ic2, ic1).should.be.true();
```

Comment:
Ah. OK. So all that normalization near the bottom of the memoizer file _was_ to make `Immutable.is` work, right?

Reply:
That's right!